cdap-ui/app/cdap/components/Reports/ReportsDetail/index.js

```diff
@@ -23,31 +23,98 @@ import ReportsStore, { ReportsActions } from 'components/Reports/store/ReportsSt
 import { Link } from 'react-router-dom';
 import {getCurrentNamespace} from 'services/NamespaceStore';
 import IconSVG from 'components/IconSVG';
+import {connect} from 'react-redux';
+import {humanReadableDate} from 'services/helpers';
 
 require('./ReportsDetail.scss');
 
-export default class ReportsDetail extends Component {
+class ReportsDetailView extends Component {
   static propTypes = {
-    match: PropTypes.object
+    match: PropTypes.object,
+    created: PropTypes.number,
+    reportName: PropTypes.string,
+    error: PropTypes.string,
+    status: PropTypes.string
   };
 
   componentWillMount() {
+    this.fetchStatus();
+  }
+
+  fetchStatus = () => {
     let params = {
       reportId: this.props.match.params.reportId
     };
 
-    MyReportsApi.getSummary(params)
-      .combineLatest(MyReportsApi.getRuns(params))
-      .subscribe(([summary, runsInfo]) => {
+    MyReportsApi.getReport(params)
+      .subscribe((res) => {
         ReportsStore.dispatch({
-          type: ReportsActions.setDetails,
+          type: ReportsActions.setInfoStatus,
           payload: {
-            runs: runsInfo.runs,
-            summary
+            info: res
           }
         });
+
+        if (res.status === 'COMPLETED') {
+          this.fetchDetails();
+        }
       });
-  }
+  };
+
+  fetchDetails = () => {
+    let params = {
+      reportId: this.props.match.params.reportId
+    };
+
+    MyReportsApi.getDetails(params)
+      .subscribe((res) => {
```

Comment:
Can we add error handlers for these API calls? Can we dispatch the error to the store?